front/src/applications/stdcm/hooks/useFetchPathProperties.ts

```diff
@@ -0,0 +1,93 @@
+// useFetchPathProperties.ts
+import { getEntities } from 'applications/editor/data/api';
+import type { TrackSectionEntity } from 'applications/editor/tools/trackEdition/types';
+import type { StdcmPathProperties } from 'applications/stdcm/types';
+import type {
+  PostInfraByInfraIdPathPropertiesApiArg,
+  PathfindingResultSuccess,
+} from 'common/api/osrdEditoastApi';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { formatSuggestedOperationalPoints } from 'modules/pathfinding/utils';
+import type { SuggestedOP } from 'modules/trainschedule/components/ManageTrainSchedule/types';
+import type { AppDispatch } from 'store';
+
+/**
+ * Custom hook that returns a function to fetch and format path properties.
+ */
+const useFetchPathProperties = () => {
+  const fetchPathProperties = async (
```

Comment:
You don't need the hook, the function can be standalone
```suggestion
/** Function to fetch and format path properties */

const fetchPathProperties = async (...) => {...}

export default fetchPathProperties;
```